### PR TITLE
Test code navigation

### LINKED ISSUE #38: Add CTRL-Click test to the typescript part of the extension
Issue splitted from https://github.com/inmanta/vscode-inmanta/issues/17

### PR DESCRIPTION
**Pull request by andraskvr**
Closes #38

### CODE REVIEW

File: src/test/suite/navigation.test.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+			} else {
+				expectedAttributeLocation = new Range(new Position(2, 11), new Position(2, 15));
+			}
+			assert.strictEqual((attributeInSameFile as Location[]).length, 1);
+			assert.strictEqual(attributeInSameFile[0].uri.fsPath, modelUri.fsPath);
+			assert.deepStrictEqual(attributeInSameFile[0].range, expectedAttributeLocation, "Attribute location in the same file doesn't match");
+
+			const typeInDifferentFile = await commands.executeCommand("vscode.executeDefinitionProvider", modelUri, new Position(4, 18));
+			assert.strictEqual((typeInDifferentFile as Location[]).length, 1);
+			assert.strictEqual(typeInDifferentFile[0].uri.fsPath, Uri.file(path.resolve(libsPath, "testmodule", "model", "_init.cf")).fsPath);
```
Comment (edvgui) [inline src/test/suite/navigation.test.ts:49]:
```suggestion
			assert.strictEqual(typeInDifferentFile[0].uri.fsPath, path.resolve(libsPath, "testmodule", "model", "_init.cf"));
```

File: src/test/suite/navigation.test.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+			assert.deepStrictEqual(typeInDifferentFile[0].range, new Range(new Position(0, 8), new Position(0, 11)), "Attribute location in different file doesn't match");
+
+			if (inmantaVersion.major < 2020 || inmantaVersion.major === 2020 && inmantaVersion.minor <= 5) {
+				console.log("Plugin navigation not available on this Inmanta version");
+			} else {
+				const pluginLocation = await commands.executeCommand("vscode.executeDefinitionProvider", modelUri, new Position(17, 15));
+				assert.strictEqual((pluginLocation as Location[]).length, 1);
+				assert.strictEqual(pluginLocation[0].uri.fsPath, Uri.file(path.resolve(libsPath, "testmodule", "plugins", "__init__.py")).fsPath);
```
Comment (edvgui) [inline src/test/suite/navigation.test.ts:57]:
```suggestion
				assert.strictEqual(pluginLocation[0].uri.fsPath, path.resolve(libsPath, "testmodule", "plugins", "__init__.py"));
```

**Comment by edvgui** [comment]
A more general comment:  
I know it has not been done in this PR first, but what do you think of the test files structure? 
It is currently like this, with multiple index-like files in the suite folder:
```
src/test/
├── navigation-workspace
│   ├── main.cf
│   └── project.yml
├── runTest.ts
├── suite
│   ├── compile.test.ts
│   ├── helpers.ts
│   ├── indexCompileTest.ts
│   ├── indexLoadExtension.ts
│   ├── indexNavTest.ts
│   ├── index.ts
│   ├── loadExtension.test.ts
│   └── navigation.test.ts
└── workspace
    ├── invalid.cf
    ├── project.yml
    └── valid.cf
```
What about rather something like this?
```
src/test/
├── runTest.ts
└── suite
    ├── compile
    │   ├── compile.test.ts
    │   ├── index.ts
    │   └── workspace
    │       ├── invalid.cf
    │       ├── libs/
    │       ├── project.yml
    │       └── valid.cf
    ├── helpers.ts
    ├── index.ts
    ├── loadExtension
    │   ├── index.ts
    │   └── loadExtension.test.ts
    └── navigation
        ├── index.ts
        ├── navigation.test.ts
        └── workspace
            ├── libs/
            ├── main.cf
            └── project.yml
```

**Review by edvgui — APPROVED** [review]

File: src/test/navigation-workspace/libs/testmodule/model/_init.cf [hunk 3]
```diff
@@ -0,0 +1,2 @@
+typedef pet as string matching
+    self == "dog" or self == "dragon" or self == "bird" or self == "dinosaur"
```
Comment (sanderr) [inline src/test/navigation-workspace/libs/testmodule/model/_init.cf:2]: Love these pet options!

**Review by sanderr — APPROVED** [review]

**Comment by sanderr** [comment]
I'm a bit too disconnected from this part to have a strong opinion on @edvgui's suggestion. It does look a bit cleaner to me.

**Pull request merged**